### ISSUE TITLE
Cannot read property 'type' of undefined when provisioning secure device.

### ISSUE THREAD

**Issue by jason-fox** [telefonicaid/iotagent-node-lib#829]
I have a docker-compose linking Keyrock, PEP Proxy and an IoT Agent:

https://github.com/jason-fox/tutorials.PEP-Proxy/blob/master/docker-compose/iot-agent.yml

The IOTA_AUTH_ENABLED is set to `true` and the attributes configured as shown to pass requests via a PEP Proxy.

```yaml
 # Orion is the context broker
  orion:
    image: fiware/orion:${ORION_VERSION}
    hostname: orion
    container_name: fiware-orion
    depends_on:
      - mongo-db
    networks:
      default:
        ipv4_address: 172.18.1.9
    expose:
      - "${ORION_PORT}"
    ports:
      - "${ORION_PORT}:${ORION_PORT}" # localhost:1026
    command: -dbhost mongo-db -logLevel DEBUG
    healthcheck:
      test: curl --fail -s http://orion:${ORION_PORT}/version || exit 1

  # IoT-Agent is configured for the UltraLight Protocol
  iot-agent:
    image: fiware/iotagent-ul:${ULTRALIGHT_VERSION}
    hostname: iot-agent
    container_name: fiware-iot-agent
    depends_on:
      - mongo-db
      - orion
      - orion-proxy
    networks:
      - default
    ports:
      - "${IOTA_NORTH_PORT}:${IOTA_NORTH_PORT}" # localhost:4041
      - "${IOTA_SOUTH_PORT}:${IOTA_SOUTH_PORT}" # localhost:7896
    environment:
      - IOTA_CB_HOST=orion-proxy # name of the context broker to update context
      - IOTA_CB_PORT=${ORION_PROXY_PORT} # port the context broker listens on to update context
      - IOTA_NORTH_PORT=${IOTA_NORTH_PORT}
      - IOTA_REGISTRY_TYPE=mongodb #Whether to hold IoT device info in memory or in a database
      - IOTA_LOG_LEVEL=DEBUG # The log level of the IoT Agent
      - IOTA_TIMESTAMP=true # Supply timestamp information with each measurement
      - IOTA_CB_NGSI_VERSION=v2 # use NGSIv2 when sending updates for active attributes
      - IOTA_AUTOCAST=true # Ensure Ultralight number values are read as numbers not strings
      - IOTA_MONGO_HOST=mongo-db # The host name of MongoDB
      - IOTA_MONGO_PORT=${MONGO_DB_PORT} # The port mongoDB is listening on
      - IOTA_MONGO_DB=iotagentul # The name of the database used in mongoDB
      - IOTA_HTTP_PORT=${IOTA_SOUTH_PORT} # The port used for device traffic over HTTP
      - IOTA_PROVIDER_URL=http://iot-agent:${IOTA_NORTH_PORT}

      - IOTA_AUTH_ENABLED=true
      - IOTA_AUTH_TYPE=oauth
      - IOTA_AUTH_HEADER=Authorization
      - IOTA_AUTH_HOST=keyrock
      - IOTA_AUTH_PORT=${KEYROCK_PORT}
      - IOTA_AUTH_URL=http://keyrock:${KEYROCK_PORT}
      - IOTA_AUTH_CLIENT_ID=tutorial-dckr-site-0000-xpresswebapp
      - IOTA_AUTH_CLIENT_SECRET=test
      - IOTA_AUTH_TOKEN_PATH=/oauth2/token

    healthcheck:
      test: curl --fail -s http://iot-agent:${IOTA_NORTH_PORT}/iot/about || exit 1

  # Keyrock is an Identity Management Front-End
  keyrock:
    image: fiware/idm:${KEYROCK_VERSION}
    container_name: fiware-keyrock
    hostname: keyrock
    networks:
      default:
        ipv4_address: 172.18.1.5
    depends_on:
      - mysql-db
    ports:
      - "${KEYROCK_PORT}:${KEYROCK_PORT}" # localhost:3005
    environment:
      - DEBUG=idm:*
      - IDM_DB_HOST=mysql-db
      - IDM_DB_PASS_FILE=/run/secrets/my_secret_data
      - IDM_DB_USER=root
      - IDM_HOST=http://localhost:${KEYROCK_PORT}
      - IDM_PORT=${KEYROCK_PORT}
      - IDM_HTTPS_PORT=${KEYROCK_HTTPS_PORT}
      - IDM_ADMIN_USER=alice
      - IDM_ADMIN_EMAIL=alice-the-admin@test.com
      - IDM_ADMIN_PASS=test
    secrets:
      - my_secret_data
    healthcheck:
      test: curl --fail -s http://localhost:${KEYROCK_PORT}/version || exit 1

  # PEP Proxy for Orion
  orion-proxy:
    image: fiware/pep-proxy:${WILMA_VERSION}
    container_name: fiware-orion-proxy
    hostname: orion-proxy
    networks:
      default:
        ipv4_address: 172.18.1.10
    depends_on:
      - keyrock
    ports:
      - "${ORION_PROXY_PORT}:${ORION_PROXY_PORT}" # localhost:1027
    expose:
      - "${ORION_PROXY_PORT}"
    environment:
      - PEP_PROXY_APP_HOST=orion
      - PEP_PROXY_APP_PORT=${ORION_PORT}
      - PEP_PROXY_PORT=${ORION_PROXY_PORT}
      - PEP_PROXY_IDM_HOST=keyrock
      - PEP_PROXY_HTTPS_ENABLED=false
      - PEP_PROXY_AUTH_ENABLED=false
      - PEP_PROXY_IDM_SSL_ENABLED=false
      - PEP_PROXY_IDM_PORT=${KEYROCK_PORT}
      - PEP_PROXY_APP_ID=tutorial-dckr-site-0000-xpresswebapp
      - PEP_PROXY_USERNAME=pep_proxy_00000000-0000-0000-0000-000000000000
      - PEP_PASSWORD=test
      - PEP_PROXY_PDP=idm
      - PEP_PROXY_MAGIC_KEY=1234
      - PEP_PROXY_PUBLIC_PATHS=/version
    healthcheck:
      test: curl --fail -s http://orion-proxy:${ORION_PROXY_PORT}/version || exit 1
```

I am able to provision a service group as shwon:

```text
op=IoTAgentNGSI.GenericMiddlewares | srv=openiot | subsrv=/ | msg=Body:
2019-11-19T11:04:16.272577800Z 
{
    "services": [
        {
            "apikey": "4jggokgpepnvsb2uv4s40d59ov",
            "cbroker": "http://orion:1027",
            "entity_type": "Thing",
            "resource": "/iot/d"
        }
    ]
}
2019-11-19T11:04:16.273000700Z 
 | comp=IoTAgent
time=2019-11-19T11:04:16.279Z | lvl=DEBUG | corr=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | trans=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | op=IoTAgentNGSI.DeviceGroupService | srv=openiot | subsrv=/ | msg=Creating new set of 1 services | comp=IoTAgent
time=2019-11-19T11:04:16.282Z | lvl=DEBUG | corr=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | trans=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | op=IoTAgentNGSI.DeviceGroupService | srv=openiot | subsrv=/ | msg=validateGroup {"apikey":"4jggokgpepnvsb2uv4s40d59ov","cbroker":"http://orion:1026","resource":"/iot/d","type":"Thing","service":"openiot","subservice":"/"} | comp=IoTAgent
time=2019-11-19T11:04:16.283Z | lvl=DEBUG | corr=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | trans=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=openiot | subsrv=/ | msg=Looking for group params ["resource","apikey"] with queryObj {"resource":"/iot/d","apikey":"4jggokgpepnvsb2uv4s40d59ov"} | comp=IoTAgent
time=2019-11-19T11:04:16.316Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Device group for fields [["resource","apikey"]] not found: [{"resource":"/iot/d","apikey":"4jggokgpepnvsb2uv4s40d59ov"}] | comp=IoTAgent
time=2019-11-19T11:04:16.316Z | lvl=ERROR | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.Alarms | srv=n/a | subsrv=n/a | msg=Raising [MONGO-ALARM]: {"name":"DEVICE_GROUP_NOT_FOUND","message":"Couldn\t find device group","code":404} | comp=IoTAgent
time=2019-11-19T11:04:16.317Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceGroupService | srv=n/a | subsrv=n/a | msg=generateDuplicateHander error [object Object] and foundGroup undefined | comp=IoTAgent
time=2019-11-19T11:04:16.331Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Storing device group with id [5dd3cc309b45da000603bf45], type [Thing], apikey [4jggokgpepnvsb2uv4s40d59ov] and resource [/iot/d] | comp=IoTAgent
time=2019-11-19T11:04:16.503Z | lvl=ERROR | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.Alarms | srv=n/a | subsrv=n/a | msg=Releasing [MONGO-ALARM] | comp=IoTAgent
time=2019-11-19T11:04:16.550Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.IOTAMService | srv=n/a | subsrv=n/a | msg=Looking for bindings for the function [configurationHandler] and protocol [null] | comp=IoTAgent
time=2019-11-19T11:04:16.621Z | lvl=DEBUG | corr=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | trans=f4a37a7a-9e7d-44bb-babb-05b768c0b7b7 | op=IoTAgentNGSI.DomainControl | srv=openiot | subsrv=/ | msg=response-time: 415 | comp=IoTAgent
```

However when I attempt to provision a **Device** - the agent fails

```text
time=2019-11-19T11:04:20.307Z | lvl=DEBUG | corr=f2953910-6047-4c80-89dc-f50b6ec96551 | trans=f2953910-6047-4c80-89dc-f50b6ec96551 | op=IoTAgentNGSI.GenericMiddlewares | srv=openiot | subsrv=/ | msg=Request for path [/iot/devices] from [localhost:4041] | comp=IoTAgent
time=2019-11-19T11:04:20.308Z | lvl=DEBUG | corr=f2953910-6047-4c80-89dc-f50b6ec96551 | trans=f2953910-6047-4c80-89dc-f50b6ec96551 | op=IoTAgentNGSI.GenericMiddlewares | srv=openiot | subsrv=/ | msg=Body:
2019-11-19T11:04:20.309926600Z 
{
    "devices": [
        {
            "device_id": "motion001",
            "entity_name": "urn:ngsd-ld:Motion:001",
            "entity_type": "Motion",
            "timezone": "Europe/Berlin",
            "attributes": [
                {
                    "object_id": "c",
                    "name": "count",
                    "type": "Integer"
                }
            ],
            "static_attributes": [
                {
                    "name": "refStore",
                    "type": "Relationship",
                    "value": "urn:ngsi-ld:Store:001"
                }
            ]
        }
    ]
}
2019-11-19T11:04:20.311213300Z 
 | comp=IoTAgent
time=2019-11-19T11:04:20.310Z | lvl=DEBUG | corr=f2953910-6047-4c80-89dc-f50b6ec96551 | trans=f2953910-6047-4c80-89dc-f50b6ec96551 | op=IoTAgentNGSI.DeviceProvisioning | srv=openiot | subsrv=/ | msg=Handling device provisioning request. | comp=IoTAgent
time=2019-11-19T11:04:20.313Z | lvl=DEBUG | corr=f2953910-6047-4c80-89dc-f50b6ec96551 | trans=f2953910-6047-4c80-89dc-f50b6ec96551 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=openiot | subsrv=/ | msg=Looking for group params ["service","subservice","type","apikey"] with queryObj {"service":"openiot","subservice":"/","type":"Motion"} | comp=IoTAgent
time=2019-11-19T11:04:20.321Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Device group for fields [["service","subservice","type","apikey"]] not found: [{"service":"openiot","subservice":"/","type":"Motion"}] | comp=IoTAgent
time=2019-11-19T11:04:20.321Z | lvl=ERROR | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.Alarms | srv=n/a | subsrv=n/a | msg=Raising [MONGO-ALARM]: {"name":"DEVICE_GROUP_NOT_FOUND","message":"Couldn\t find device group","code":404} | comp=IoTAgent
time=2019-11-19T11:04:20.322Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.BidirectionalPlugin | srv=n/a | subsrv=n/a | msg=Extracting attribute list | comp=IoTAgent
time=2019-11-19T11:04:20.323Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.BidirectionalPlugin | srv=n/a | subsrv=n/a | msg=Sending bidirectionality subscriptions for device [motion001] | comp=IoTAgent
time=2019-11-19T11:04:20.324Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.IOTAMService | srv=n/a | subsrv=n/a | msg=Looking for bindings for the function [deviceProvisioningHandler] and protocol [null] | comp=IoTAgent
time=2019-11-19T11:04:20.325Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.IOTAMService | srv=n/a | subsrv=n/a | msg=Binding found for function [deviceProvisioningHandler] and protocol [null] | comp=IoTAgent
time=2019-11-19T11:04:20.328Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBDeviceRegister | srv=n/a | subsrv=n/a | msg=Looking for device with id [motion001]. | comp=IoTAgent
time=2019-11-19T11:04:20.333Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBDeviceRegister | srv=n/a | subsrv=n/a | msg=Device [motion001] not found. | comp=IoTAgent
time=2019-11-19T11:04:20.335Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Looking for group params ["service","subservice","type","apikey"] with queryObj {"service":"openiot","subservice":"/","type":"Motion"} | comp=IoTAgent
time=2019-11-19T11:04:20.338Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Device group for fields [["service","subservice","type","apikey"]] not found: [{"service":"openiot","subservice":"/","type":"Motion"}] | comp=IoTAgent
time=2019-11-19T11:04:20.339Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=deviceData after merge with conf: {"id":"motion001","type":"Motion","name":"urn:ngsd-ld:Motion:001","service":"openiot","subservice":"/","active":[{"object_id":"c","name":"count","type":"Integer"}],"staticAttributes":[{"name":"refStore","type":"Relationship","value":"urn:ngsi-ld:Store:001"}],"timezone":"Europe/Berlin","transport":"HTTP","internalId":null,"subscriptions":[],"polling":true} | comp=IoTAgent
time=2019-11-19T11:04:20.345Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=deviceData before merge with conf: {"id":"motion001","type":"Motion","name":"urn:ngsd-ld:Motion:001","service":"openiot","subservice":"/","active":[{"object_id":"c","name":"count","type":"Integer"}],"staticAttributes":[{"name":"refStore","type":"Relationship","value":"urn:ngsi-ld:Store:001"}],"lazy":null,"commands":[],"timezone":"Europe/Berlin","transport":"HTTP","internalId":null,"subscriptions":[],"polling":true} | comp=IoTAgent
time=2019-11-19T11:04:20.347Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=Registering device into NGSI Service:
{
    "id": "motion001",
    "type": "Motion",
    "name": "urn:ngsd-ld:Motion:001",
    "service": "openiot",
    "subservice": "/",
    "active": [
        {
            "object_id": "c",
            "name": "count",
            "type": "Integer"
        }
    ],
    "staticAttributes": [
        {
            "name": "refStore",
            "type": "Relationship",
            "value": "urn:ngsi-ld:Store:001"
        }
    ],
    "lazy": null,
    "commands": [],
    "timezone": "Europe/Berlin",
    "transport": "HTTP",
    "internalId": null,
    "subscriptions": [],
    "polling": true
} | comp=IoTAgent
time=2019-11-19T11:04:20.349Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=Registration with Context Provider is not needed.Device without lazy atts or commands | comp=IoTAgent
time=2019-11-19T11:04:20.351Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=deviceData: {"id":"motion001","type":"Motion","name":"urn:ngsd-ld:Motion:001","service":"openiot","subservice":"/","active":[{"object_id":"c","name":"count","type":"Integer"}],"staticAttributes":[{"name":"refStore","type":"Relationship","value":"urn:ngsi-ld:Store:001"}],"lazy":null,"commands":[],"timezone":"Europe/Berlin","transport":"HTTP","internalId":null,"subscriptions":[],"polling":true} | comp=IoTAgent
time=2019-11-19T11:04:20.353Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=config.timestamp undefined true | comp=IoTAgent
time=2019-11-19T11:04:20.361Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=Creating initial entity in the Context Broker:
 {
    "url": "http://orion-proxy:1027/v2/entities?options=upsert",
    "method": "POST",
    "json": {
        "id": "urn:ngsd-ld:Motion:001",
        "type": "Motion",
        "count": {
            "type": "Integer",
            "value": " "
        },
        "refStore": {
            "type": "Relationship",
            "value": "urn:ngsi-ld:Store:001"
        },
        "TimeInstant": {
            "type": "DateTime",
            "value": "2019-11-19T11:04:20.356Z"
        }
    },
    "headers": {
        "fiware-service": "openiot",
        "fiware-servicepath": "/",
        "fiware-correlator": "c1918726-6142-47d4-bf32-680c40801197"
    }
} | comp=IoTAgent
time=2019-11-19T11:04:20.362Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=executeWithSecurity | comp=IoTAgent
time=2019-11-19T11:04:20.362Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Looking for group params ["type"] with queryObj {"type":"Motion"} | comp=IoTAgent
time=2019-11-19T11:04:20.366Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.MongoDBGroupRegister | srv=n/a | subsrv=n/a | msg=Device group for fields [["type"]] not found: [{"type":"Motion"}] | comp=IoTAgent
time=2019-11-19T11:04:20.366Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DeviceService | srv=n/a | subsrv=n/a | msg=error {"name":"DEVICE_GROUP_NOT_FOUND","message":"Couldn\t find device group","code":404} in get group device | comp=IoTAgent
time=2019-11-19T11:04:20.368Z | lvl=ERROR | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DomainControl | srv=n/a | subsrv=n/a | msg={ TypeError: Cannot read property 'type' of undefined
    at Function.<anonymous> (/opt/iotaul/node_modules/iotagent-node-lib/lib/services/devices/deviceService.js:302:84)
    at /opt/iotaul/node_modules/iotagent-node-lib/lib/services/common/alarmManagement.js:105:29
    at handleGet (/opt/iotaul/node_modules/iotagent-node-lib/lib/services/groups/groupRegistryMongoDB.js:233:17)
    at /opt/iotaul/node_modules/mongoose/lib/model.js:4598:16
    at /opt/iotaul/node_modules/mongoose/lib/utils.js:264:16
    at _hooks.execPost (/opt/iotaul/node_modules/mongoose/lib/query.js:4326:11)
    at /opt/iotaul/node_modules/kareem/index.js:135:16
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickDomainCallback (internal/process/next_tick.js:219:9)
  domain: 
   Domain {
     domain: null,
     _events: { error: [Function: domainErrorHandler] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [],
     start: 1574161335687,
     trans: 'c1918726-6142-47d4-bf32-680c40801197',
     corr: 'c1918726-6142-47d4-bf32-680c40801197',
     op: 'IoTAgentNGSI.MongoDBGroupRegister' },
  domainThrown: true } | comp=IoTAgent
time=2019-11-19T11:04:20.381Z | lvl=DEBUG | corr=c1918726-6142-47d4-bf32-680c40801197 | trans=c1918726-6142-47d4-bf32-680c40801197 | op=IoTAgentNGSI.DomainControl | srv=n/a | subsrv=n/a | msg=response-time: 124694 | comp=IoTAgent
```

Do you have any idea what could be causing the **TypeError: Cannot read property 'type' of undefined**?

**Comment by fgalan** [comment]
A couple of tests that could be done and could bring some light into this case:

* Reset scenario (clean up databases, basically), then skip the service provision part. I mean, do only the device provising step.
* Reset scenario (clean up databases, basically), then do the service provision, then do the device provision but using the same entity_type than the service, i.e. changing from `"entity_type": "Motion"` to `"entity_type": "Thing"`

In addition, what about removign IDM and Kyrock (for debugging)? Do you get the same results (probably, yes, but let's ensure)?

**Comment by fgalan** [comment]
Fixed by PR #833
